### PR TITLE
[State Sync] Add additional tests for subscription streams.

### PR DESCRIPTION
**Pull request by JoshLind**
### Description
This PR builds on https://github.com/aptos-labs/aptos-core/pull/9921 and adds additional unit and integration tests for subscription streams. Note: no other logical changes should take place.

### Test Plan
New and existing test infrastructure.

### CODE REVIEW

File: state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs [hunk 1]
```diff
@@ -812,6 +1017,26 @@ fn create_range_random_u64(min_value: u64, max_value: u64) -> u64 {
     rng.gen_range(min_value, max_value)
 }
 
+/// Determines the target ledger info for the given known epoch
+async fn determine_targer_ledger_info(
```
Comment (bchocho) [inline state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs:1021]: typo: targer -> target

File: state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs [hunk 2]
```diff
@@ -611,39 +651,204 @@ impl AptosDataClientInterface for MockAptosDataClient {
...
+            DataRequest::SubscribeTransactionsWithProof(SubscribeTransactionsWithProofRequest {
+                subscription_stream_metadata: SubscriptionStreamMetadata {
+                    known_version_at_stream_start,
+                    known_epoch_at_stream_start,
+                    subscription_stream_id: subscription_request_metadata.subscription_stream_id,
+                },
+                subscription_stream_index,
+                include_events,
+            });
+        self.verify_request_timeout(request_timeout_ms, false, true, data_request);
```
Comment (bchocho) [inline state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs:750]: nit: would it be clearer as `verify_request_timeout_value`? At first glance I thought a timeout was expected to be triggered.

File: state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs [hunk 3]
```diff
@@ -133,24 +142,40 @@ impl MockAptosDataClient {
...
+
+        aptos_data_client
+    }
+
+    /// Emulates network latencies by sleeping for 10 -> 50 ms
     fn emulate_network_latencies(&self) {
-        if self.skip_emulate_network_latencies {
-            return;
+        if !self.skip_emulate_network_latencies {
+            thread::sleep(Duration::from_millis(create_range_random_u64(10, 50)));
```
Comment (bchocho) [inline state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs:157]: could this be tokio sleep? presumable other tasks could run while waiting for network

File: state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs (hunk 1 above)
Comment (JoshLind) [reply state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs:1021]: Aah, thanks 😄

File: state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs (hunk 2 above)
Comment (JoshLind) [reply state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs:750]: Yeah, makes sense!

File: state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs (hunk 3 above)
Comment (JoshLind) [reply state-sync/state-sync-v2/data-streaming-service/src/tests/utils.rs:157]: Yeah, makes sense. 😄 I've made them `async` functions, updated them to use `tokio::time::sleep` and also updated the `emulate_network_request_timeout` method, too.

**Review by bchocho — APPROVED** [review]
LGTM

**Review by msmouse — APPROVED** [review]

**Pull request merged**